app/components/ellipsisDropdownMenu.tsx

```diff
@@ -1,21 +1,25 @@
-import { Fragment } from "react";
-import { Menu, Transition } from "@headlessui/react";
+import { Fragment, ReactNode } from "react";
+import { Menu,MenuItem, MenuButton, MenuItems, Transition } from "@headlessui/react";
 import React from "react";
-import Link from "next/link";
 
 interface DropdownMenuProps {
-    options: { label: string, href: string, show?: boolean }[];
+    options: {  component:()=>ReactNode,show?: boolean }[];
```

Comment:
Just change the props here to apply a component without func. This comment related to my comments below.

Reply:
fixed